corgi/tasks/brew.py

```diff
@@ -382,3 +421,12 @@ def load_brew_tags() -> None:
                 if created:
                     no_of_created += 1
             logger.info("Saving %s new builds for %s", no_of_created, brew_tag)
+
+
+def fetch_modular_builds(relations_query):
+    build_ids: set[int] = set()
+    for build_id in relations_query:
+        build_id_int = int(build_id)
+        build_ids.add(build_id_int)
+        fetch_modular_build.delay(build_id_int)
+    return list(build_ids)
```

Comment:
Not related to this MR specifically, but this could be cleaned up:
```suggestion
def fetch_modular_builds(relations_query):
    build_ids = list(relations_query)
    for build_id in build_ids:
        fetch_modular_build.delay(build_id)
    return build_ids
```
I thought at first this code had a bug, where we might call fetch_modular_build() multiple times if the same build_id appeared multiple times in relations query.

But actually, all the code I see in rhel_compose.py that calls fetch_modular_build() already uses .values_list("build_id", flat=True).distinct(), so there's no need to convert the build_ids to a set. We already get a distinct queryset of the build IDs, so we can just convert that to a list and iterate through it, instead of building a set which would be slower and take more memory.

Reply:
I updated this as you suggested and return the original queryset as well.

Reply:
I actually made another improvement here. What was happening with the original code here was the fetch_modular_build was being called with all build_ids and then only slow_fetch_brew_build was being called, which meant that modular builds where 'half baked' until the whole stream was processed. I improved it by calling slow_fetch_brew_brew after fetch_modular_build for each build_id. That also means I don't need to return anything from fetch_modular_builds function.